assets/app/scripts/services/apiEndpoint.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('ApiEndpointService', function(API_CFG) {
+
+    // TODO: prob should not modify the config data,
+    // instead keep this as local vars?
+    // var defaultVersion = { os: 'v1', k8s: 'v1' }
+    API_CFG.openshift.defaultVersion = "v1";
+    API_CFG.k8s.defaultVersion = "v1";
+
+    // generates something like:
+    // https://localhost:8443
+    var openshiftAPIBaseUrl = function() {
+      var protocol = window.location.protocol === "http:" ? "http" : "https";
+      var hostPort = API_CFG.openshift.hostPort;
+      return new URI({protocol: protocol, hostname: hostPort}).toString();
+    };
+
+    var normalizeResource = function(resource) {
+       if (!resource) {
+        return;
+       }
+
+       // only lowercase the first segment, leaving subresources as-is (some are case-sensitive)
+       var segments = resource.split("/");
+       segments[0] = segments[0].toLowerCase();
+       var normalized = segments.join("/");
+       if (resource !== normalized) {
+         Logger.warn('Non-lower case resource "' + resource + '"');
+       }
+
+       return normalized;
+    };
+
+    // port of restmapper.go#kindToResource
+    var kindToResource = function(kind) {
```

Comment:
while you're in here, update to the latest impl (https://github.com/kubernetes/kubernetes/blob/47d2f4168009f469fa842f978d4c4466e7d4b186/pkg/api/meta/restmapper.go#L132)